sockeye/inference.py

```diff
@@ -862,17 +912,22 @@ def _get_inference_input(self,
         for j, trans_input in enumerate(trans_inputs):
             num_tokens = len(trans_input)  # includes eos
             max_output_lengths.append(self._get_max_output_length(num_tokens))
-            source[j, :num_tokens, 0] = tokens2ids(trans_input.tokens, self.source_vocabs[0])
-
+            num_tokens = num_tokens + trans_input.num_source_prefix_tokens()
+            source[j, :num_tokens, 0] = tokens2ids(trans_input.get_source_prefix_tokens() \
```

Comment:
we could avoid the explicit construction of the concatenated list with `chain(trans_input.get_source_prefix_tokens(), trans_input.tokens)`